pipelines/folioRancher/folioDevTools/moduleDeployment/deployModuleFromFeatureBranchEureka/Jenkinsfile

```diff
@@ -0,0 +1,268 @@
+properties([
+  buildDiscarder(logRotator(numToKeepStr: '30')),
+  disableConcurrentBuilds(),
+  parameters([
+    folioParameters.moduleName(),
+    folioParameters.branchWithRef('MODULE_BRANCH', 'MODULE_NAME'),
+    string(name: 'MAVEN_ARGS', defaultValue: '-DskipTests', description: 'Maven build arguments'),
+    folioParameters.cluster(),
+    folioParameters.namespace(),
+    folioParameters.configType(),
+    folioParameters.agent(),
+    folioParameters.refreshParameters(),
+  ])
+])
+
+if (params.REFRESH_PARAMETERS) {
+  currentBuild.result = 'ABORTED'
+  return
+}
+
+
+/** Job variables */
+Logger logger = new Logger(this, env.JOB_BASE_NAME)
+
+/** Collect Rancher Namespace Configuration */
+EurekaNamespace namespace = new EurekaNamespace(params.CLUSTER, params.NAMESPACE)
+  .withDeploymentConfigType(params.CONFIG_TYPE) as EurekaNamespace
+
+/** Assign Desired Resource Profile for Environment */
+namespace.addDeploymentConfig(folioTools.getPipelineBranch())
+
+/** Assign Default Tenant Id to Namespace instance */
+String defaultTenantId = 'diku'
+namespace.withDefaultTenant(defaultTenantId)
+
+/** Collect Folio module configuration */
+FolioModule module = new FolioModule()
+module.name = params.MODULE_NAME
+
+/** Collect feature branch and hash */
+String featureBranch = params.MODULE_BRANCH
+String featureHash = new GitHubClient(this).getBranchInfo(module.getName(), featureBranch).commit.sha.take(7)
+
+/** Collect Maven arguments */
+String mavenArguments = params.MAVEN_ARGS.trim()
+
+/** Init Eureka Application Instance */
+Eureka eureka = new Eureka(this, namespace.generateDomain('kong'), namespace.generateDomain('keycloak'), false)
+
+/** Pipeline */
+ansiColor('xterm') {
+  // Common Gradle Options
+  String commonGradleOpts = '--quiet --console plain --no-daemon'
+
+  node(params.AGENT) {
+    try {
+      stage('Ini') {
+        buildName "#${params.MODULE_NAME}.${env.BUILD_ID}"
+        buildDescription "Env: ${namespace.getWorkspaceName()}\nBranch: ${params.MODULE_BRANCH}\n" + "Config: ${params.CONFIG_TYPE}\n${currentBuild.getBuildCauses()[0].shortDescription}"
+      }
+
+      stage('[Git] Checkout module source') {
+        checkout([
+          $class           : 'GitSCM',
+          branches         : [[name: "*/${featureBranch}"]],
+          extensions       : [[$class: 'RelativeTargetDirectory', relativeTargetDir: module.name],
+                              [$class: 'CleanBeforeCheckout', deleteUntrackedNestedRepositories: true],
+                              [$class: 'AuthorInChangelog'],
+                              [$class: 'SubmoduleOption', recursiveSubmodules: true]],
+          userRemoteConfigs: [[url: "https://github.com/folio-org/${module.name}.git"]]
+        ])
+      }
+
+      stage('[Maven/Gradle] Compile') {
+        // Check if Maven or Gradle build tool is used
+        module.buildTool = fileExists("${module.name}/pom.xml") ? 'maven' : 'gradle'
+
+        if (module.buildTool == 'maven') {
```

Comment:
It is will looks better if change 'if' to swtich/case and also move common(repeatable) code to separate function. Even inside this Jenkinsfile

Reply:
Code adjusted accordingly to your suggestion.
Please take a look on changes.